src/main/java/seedu/address/model/property/Property.java

```diff
@@ -60,57 +64,70 @@ public Address getAddress() {
         return Collections.unmodifiableSet(tags);
     }
 
+    public Seller getSeller() {
+        return seller;
+    }
+
+    public Price getPrice() {
+        return price;
+    }
+
     /**
-     * Returns true if both persons have the same name.
-     * This defines a weaker notion of equality between two persons.
+     * Returns true if both properties have the same name.
+     * This defines a weaker notion of equality between two properties.
      */
-    public boolean isSamePerson(Person otherPerson) {
-        if (otherPerson == this) {
+    public boolean isSameProperty(Property otherProperty) {
+        if (otherProperty == this) {
             return true;
         }
 
-        return otherPerson != null
-                && otherPerson.getName().equals(getName());
+        return otherProperty != null
+                && otherProperty.getName().equals(getName());
     }
 
     /**
-     * Returns true if both persons have the same identity and data fields.
-     * This defines a stronger notion of equality between two persons.
+     * Returns true if both properties have the same identity and data fields.
+     * This defines a stronger notion of equality between two properties.
      */
     @Override
     public boolean equals(Object other) {
         if (other == this) {
             return true;
         }
 
-        if (!(other instanceof Person)) {
+        if (!(other instanceof Property)) {
             return false;
         }
 
-        Person otherPerson = (Person) other;
-        return otherPerson.getName().equals(getName())
-                && otherPerson.getPhone().equals(getPhone())
-                && otherPerson.getEmail().equals(getEmail())
-                && otherPerson.getAddress().equals(getAddress())
-                && otherPerson.getTags().equals(getTags());
+        Property otherProperty = (Property) other;
+        return otherProperty.getName().equals(getName())
+                && otherProperty.getPhone().equals(getPhone())
+                && otherProperty.getSeller().equals(getPhone())
+                && otherProperty.getEmail().equals(getEmail())
+                && otherProperty.getAddress().equals(getAddress())
+                && otherProperty.getTags().equals(getTags());
     }
 
     @Override
     public int hashCode() {
         // use this method for custom fields hashing instead of implementing your own
-        return Objects.hash(name, phone, email, address, tags);
+        return Objects.hash(name, phone, email, address, tags, seller);
```

Comment:
what about the price?

Reply:
IMO Price shouldn't be tied to identity of the Property, i.e. two properties with everything identical except price should still be considered identical.

Reply:
I've made the change to include price.